src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,49 @@
+import { FC, memo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+export interface TodoListProps {
+  todos: Todo[],
+  onDeleteTodo: (todoId: number) => void;
+  tempTodo: Todo | null,
+  deletingTodoIds: number[],
+  updateTodo:(
+    todoId: number,
+    fieldsToUpdate: Partial<Pick<Todo, 'title' | 'completed'>>
+  ) => Promise<void>,
+  updatingTodosIds: number[],
+}
+
+export const TodoList: FC<TodoListProps> = memo(({
+  todos,
+  onDeleteTodo,
+  tempTodo,
+  deletingTodoIds,
+  updateTodo,
+  updatingTodosIds,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <TodoItem
+          key={todo.id}
+          todo={todo}
+          onDeleteTodo={onDeleteTodo}
+          isDeleting={deletingTodoIds.includes(todo.id)}
+          updateTodo={updateTodo}
+          isUpdating={updatingTodosIds.includes(todo.id)}
```

Comment:
Maybe let's calculate `includes` one time with `useMemo` ?

Reply:
Thank you for comments, will take it into account